earth_enterprise/src/fusion/autoingest/sysman/AssetD.h

```diff
@@ -56,4 +62,26 @@ class AssetImplD : public virtual AssetImpl
 typedef DerivedAssetHandle_<Asset, AssetImplD> AssetD;
 typedef MutableAssetHandleD_<AssetD> MutableAssetD;
 
+template<typename MutableVersionHandleType, typename MutableAssetHandleType, typename ConfigType>
+MutableVersionHandleType AssetImplD::MakeNewVersion(MutableAssetHandleType &asset, const ConfigType &config)
+{
+    typedef typename MutableVersionHandleType::Impl VerImplType;
+    MutableVersionHandleType newver(std::make_shared<VerImplType>
+                                      (asset.operator->(), config));
+
+    asset->AddVersionRef(newver->GetRef());
```

Comment:
I guess there's a concern here where MutableVersionHandleType and MutableAssetHandleType don't necessarily have to match and it would still compile. I can live with that, though.